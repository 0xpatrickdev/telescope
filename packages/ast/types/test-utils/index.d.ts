import { ProtoType } from '@osmonauts/types';
import { ProtoStore } from '@osmonauts/proto-parser';
import { AminoParseContext } from '../src/encoding/context';
export declare const expectCode: (ast: any) => void;
export declare const printCode: (ast: any) => void;
export declare const prepareContext: (store: ProtoStore, protoFile: string) => {
    context: AminoParseContext;
    root: import("@osmonauts/types").ProtoRoot;
    protos: ProtoType[];
};
