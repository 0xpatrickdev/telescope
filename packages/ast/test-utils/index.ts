import generate from '@babel/generator';
import { ProtoType } from '@osmonauts/types';
import { ProtoStore, getNestedProto } from '@osmonauts/proto-parser'
import { AminoParseContext } from '../src/encoding/context';
import { GenericParseContext } from '../src/encoding';
import { defaultTelescopeOptions } from '@osmonauts/types';

export const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}
export const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}

export const prepareContext = (store: ProtoStore, protoFile: string) => {
    const ref = store.findProto(protoFile);
    const traversed = ref.traversed
    if (!traversed) {
        throw new Error('not traversed!');
    }
    const proto: Record<string, ProtoType> = getNestedProto(traversed);
    const protos: ProtoType[] = Object.values(proto).filter(
        proto => proto.name.startsWith('Msg')
            &&
            !proto.name.endsWith('Response')
            &&
            proto.name !== 'Msg'
    );

    const context = new AminoParseContext(ref, store, defaultTelescopeOptions);

    return {
        context,
        root: traversed,
        protos
    };
};

export const getGenericParseContext = () => {
    const context = new GenericParseContext(null, null, defaultTelescopeOptions);
    return context;
};