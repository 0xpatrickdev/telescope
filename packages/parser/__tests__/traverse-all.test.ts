import { getTestProtoStore } from '../test-utils';
const store = getTestProtoStore();

describe('osmosis/gamm/v1beta1/tx', () => {
  store.traverseAll();
  const ref = store.findProto('osmosis/gamm/v1beta1/tx.proto');

  it('Coin', () => {
    const Coin = store.get(ref, 'Coin');
    expect(Coin.importType).toEqual('import')
    expect(Coin.obj.fields).toMatchSnapshot();
  })
  it('SwapAmountInRoute', () => {
    const SwapAmountInRoute = store.get(ref, 'SwapAmountInRoute');
    expect(SwapAmountInRoute.importType).toEqual('local')
    expect(SwapAmountInRoute.obj.fields).toMatchSnapshot();
  })
});

describe('google/api/expr/conformance/v1alpha1/conformance_service', () => {
  store.traverseAll();
  const ref = store.findProto('google/api/expr/conformance/v1alpha1/conformance_service.proto');

  it('Decl', () => {
    const Decl = store.get(ref, 'Decl');
    expect(Decl.importType).toEqual('import')
    expect(Decl.obj.fields).toMatchSnapshot();
  })
  it('ParsedExpr', () => {
    const ParsedExpr = store.get(ref, 'ParsedExpr');
    expect(ParsedExpr.importType).toEqual('import')
    expect(ParsedExpr.obj.fields).toMatchSnapshot();
  })

  it('CheckRequest', () => {
    const CheckRequest = store.get(ref, 'CheckRequest');
    expect(CheckRequest.importType).toEqual('local')
    expect(CheckRequest.obj.fields).toMatchSnapshot();
  })

  it('IdentDecl', () => {
    const CheckRequest = store.get(ref, 'CheckRequest');

    // nested could look like this...
    const ref2 = store.findProto(CheckRequest.obj.fields.typeEnv.import)
    const IdentDecl = store.get(ref2, 'IdentDecl');
    expect(IdentDecl.importType).toEqual('local')
    expect(IdentDecl.obj.fields).toMatchSnapshot();
  })
});

