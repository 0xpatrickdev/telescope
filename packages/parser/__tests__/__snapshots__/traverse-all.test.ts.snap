// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`google/api/expr/conformance/v1alpha1/conformance_service CheckRequest 1`] = `
Object {
  "container": Object {
    "comment": "The protocol buffer context.  See \\"Name Resolution\\" in the
Language Definition.",
    "id": 3,
    "isScalar": true,
    "options": Object {
      "(telescope:camel)": "container",
      "(telescope:name)": "container",
      "(telescope:orig)": "container",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
  "noStdEnv": Object {
    "comment": "If true, use only the declarations in [type_env][google.api.expr.conformance.v1alpha1.CheckRequest.type_env].  If false (default),
add declarations for the standard definitions to the type environment.  See
\\"Standard Definitions\\" in the Language Definition.",
    "id": 4,
    "isScalar": true,
    "options": Object {
      "(telescope:camel)": "noStdEnv",
      "(telescope:name)": "noStdEnv",
      "(telescope:orig)": "no_std_env",
    },
    "parsedType": Object {
      "name": "bool",
      "type": "native",
    },
    "type": "bool",
    "typeNum": 12,
  },
  "parsedExpr": Object {
    "comment": "Required. The parsed representation of the CEL program.",
    "id": 1,
    "import": "google/api/expr/v1alpha1/syntax.proto",
    "importedName": "google.api.expr.v1alpha1.ParsedExpr",
    "options": Object {
      "(telescope:camel)": "parsedExpr",
      "(telescope:name)": "parsedExpr",
      "(telescope:orig)": "parsed_expr",
    },
    "parsedType": Object {
      "name": "ParsedExpr",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
    ],
    "scopeType": "protoImport",
    "type": "google.api.expr.v1alpha1.ParsedExpr",
  },
  "typeEnv": Object {
    "comment": "Declarations of types for external variables and functions.
Required if program uses external variables or functions
not in the default environment.",
    "id": 2,
    "import": "google/api/expr/v1alpha1/checked.proto",
    "importedName": "google.api.expr.v1alpha1.Decl",
    "options": Object {
      "(telescope:camel)": "typeEnv",
      "(telescope:name)": "typeEnv",
      "(telescope:orig)": "type_env",
    },
    "parsedType": Object {
      "name": "Decl",
      "type": "Type",
    },
    "rule": "repeated",
    "scope": Array [
      "google.api.expr.v1alpha1",
    ],
    "scopeType": "protoImport",
    "type": "google.api.expr.v1alpha1.Decl",
  },
}
`;

exports[`google/api/expr/conformance/v1alpha1/conformance_service Decl 1`] = `
Object {
  "function": Object {
    "comment": "Function declaration.",
    "id": 3,
    "options": Object {
      "(telescope:camel)": "function",
      "(telescope:name)": "function",
      "(telescope:orig)": "function",
    },
    "parsedType": Object {
      "name": "FunctionDecl",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
      "Decl",
    ],
    "type": "FunctionDecl",
  },
  "ident": Object {
    "comment": "Identifier declaration.",
    "id": 2,
    "options": Object {
      "(telescope:camel)": "ident",
      "(telescope:name)": "ident",
      "(telescope:orig)": "ident",
    },
    "parsedType": Object {
      "name": "IdentDecl",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
      "Decl",
    ],
    "type": "IdentDecl",
  },
  "name": Object {
    "comment": "The fully qualified name of the declaration.

Declarations are organized in containers and this represents the full path
to the declaration in its container, as in \`google.api.expr.Decl\`.

Declarations used as [FunctionDecl.Overload][google.api.expr.v1alpha1.Decl.FunctionDecl.Overload] parameters may or may not
have a name depending on whether the overload is function declaration or a
function definition containing a result [Expr][google.api.expr.v1alpha1.Expr].",
    "id": 1,
    "isScalar": true,
    "options": Object {
      "(telescope:camel)": "name",
      "(telescope:name)": "name",
      "(telescope:orig)": "name",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
}
`;

exports[`google/api/expr/conformance/v1alpha1/conformance_service IdentDecl 1`] = `
Object {
  "doc": Object {
    "comment": "Documentation string for the identifier.",
    "id": 3,
    "isScalar": true,
    "options": Object {
      "(telescope:camel)": "doc",
      "(telescope:name)": "doc",
      "(telescope:orig)": "doc",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
  "type": Object {
    "comment": "Required. The type of the identifier.",
    "id": 1,
    "options": Object {
      "(telescope:camel)": "type",
      "(telescope:name)": "type",
      "(telescope:orig)": "type",
    },
    "parsedType": Object {
      "name": "Type",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
    ],
    "type": "Type",
  },
  "value": Object {
    "comment": "The constant value of the identifier. If not specified, the identifier
must be supplied at evaluation time.",
    "id": 2,
    "import": "google/api/expr/v1alpha1/syntax.proto",
    "importedName": "Constant",
    "options": Object {
      "(telescope:camel)": "value",
      "(telescope:name)": "value",
      "(telescope:orig)": "value",
    },
    "parsedType": Object {
      "name": "Constant",
      "type": "Type",
    },
    "scope": Array [
      Array [
        "google.api.expr.v1alpha1",
      ],
    ],
    "scopeType": "import",
    "type": "Constant",
  },
}
`;

exports[`google/api/expr/conformance/v1alpha1/conformance_service ParsedExpr 1`] = `
Object {
  "expr": Object {
    "comment": "The parsed expression.",
    "id": 2,
    "options": Object {
      "(telescope:camel)": "expr",
      "(telescope:name)": "expr",
      "(telescope:orig)": "expr",
    },
    "parsedType": Object {
      "name": "Expr",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
    ],
    "type": "Expr",
  },
  "sourceInfo": Object {
    "comment": "The source info derived from input that generated the parsed \`expr\`.",
    "id": 3,
    "options": Object {
      "(telescope:camel)": "sourceInfo",
      "(telescope:name)": "sourceInfo",
      "(telescope:orig)": "source_info",
    },
    "parsedType": Object {
      "name": "SourceInfo",
      "type": "Type",
    },
    "scope": Array [
      "google.api.expr.v1alpha1",
    ],
    "type": "SourceInfo",
  },
}
`;

exports[`osmosis/gamm/v1beta1/tx Coin 1`] = `
Object {
  "amount": Object {
    "comment": null,
    "id": 2,
    "isScalar": true,
    "options": Object {
      "(cosmos_proto.scalar)": "cosmos.Int",
      "(gogoproto.customtype)": "Int",
      "(gogoproto.nullable)": false,
      "(telescope:camel)": "amount",
      "(telescope:name)": "amount",
      "(telescope:orig)": "amount",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
  "denom": Object {
    "comment": null,
    "id": 1,
    "isScalar": true,
    "options": Object {
      "(telescope:camel)": "denom",
      "(telescope:name)": "denom",
      "(telescope:orig)": "denom",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
}
`;

exports[`osmosis/gamm/v1beta1/tx SwapAmountInRoute 1`] = `
Object {
  "poolId": Object {
    "comment": null,
    "id": 1,
    "isScalar": true,
    "options": Object {
      "(gogoproto.moretags)": "yaml:pool_id",
      "(telescope:camel)": "poolId",
      "(telescope:name)": "poolId",
      "(telescope:orig)": "poolId",
    },
    "parsedType": Object {
      "name": "uint64",
      "type": "native",
    },
    "type": "uint64",
    "typeNum": 8,
  },
  "tokenOutDenom": Object {
    "comment": null,
    "id": 2,
    "isScalar": true,
    "options": Object {
      "(gogoproto.moretags)": "yaml:token_out_denom",
      "(telescope:camel)": "tokenOutDenom",
      "(telescope:name)": "tokenOutDenom",
      "(telescope:orig)": "tokenOutDenom",
    },
    "parsedType": Object {
      "name": "string",
      "type": "native",
    },
    "type": "string",
    "typeNum": 13,
  },
}
`;
